### PR TITLE
Ignore getting classpath from SystemClassLoader on Java 9+

### PR DESCRIPTION
**Pull request by zcai1**
`ClassLoader.getSystemClassLoader()` no longer returns `URLClassLoader` starting from Java 9+, so this piece of code could potentially throw `ClassCastException`.

Unfortunately, I couldn't find a nice alternative approach, so it is left as a TODO.

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java [hunk 1]
```diff
@@ -2367,9 +2367,15 @@ private void logBugInCF(BugInCF ce) {
             if (printClasspath) {
                 msg.add("Classpath:");
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java:2368]: As this code is building a diagnostic message, you should try to make it fail more gracefully.
How about moving this into the `then` part and adding an `else` part that mentions that Java 9+ is used?
The comment explaining the TODO should then move into the else part.

Comment (zcai1) [reply framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java]: feedback addressed

**Comment by mernst** [comment]
There is something strange about this branch.
I cannot push commits to the branch.  (@zcai1 Is the "allow commits from maintainers" checkbox checked?)
I have made a pull request against the branch.  That pull request will be easier to understand if you pull master into the branch first.
@zcai1 Please try to avoid force-pushing (though I don't have evidence that that is the problem).

**Comment by zcai1** [comment]
> There is something strange about this branch.
> I cannot push commits to the branch. (@zcai1 Is the "allow commits from maintainers" checkbox checked?)
> I have made a pull request against the branch. That pull request will be easier to understand if you pull master into the branch first.
> @zcai1 Please try to avoid force-pushing (though I don't have evidence that that is the problem).

Thank you for your feedback! The PR has been merged.

The "allow commits from maintainers" is always checked, so I'm not sure why it didn't work.

**Comment by mernst** [comment]
> The "allow commits from maintainers" is always checked, so I'm not sure why it didn't work.

That's what I figured.  Maybe the `+` in the branch name was a problem?  Anyway, the pull request looks good now.  Thanks again.

**Pull request merged**